pkg/providers/vsphere/config/secret.yaml

```diff
@@ -7,6 +7,10 @@ type: kubernetes.io/basic-auth
 data:
   username: {{.vsphereUsername | b64enc}}
   password: {{.vspherePassword | b64enc}}
+  usernameCSI: "{{.eksaCSIUsername | b64enc}}"
+  passwordCSI: "{{.eksaCSIPassword | b64enc}}"
+  usernameCP: "{{.eksaCloudProviderUsername | b64enc}}"
+  passwordCP: "{{.eksaCloudProviderPassword | b64enc}}"
```

Comment:
what are these for?

Reply:
This seems to be a little known [feature](https://anywhere.eks.amazonaws.com/docs/reference/clusterspec/vsphere/#optional-vsphere-credentials) of ours that allows developers to provide different sets of vSphere creds for cluster management, the CSI driver, and the vSphere Cloud Provider.

Reply:
Oh sorry, I meant why are we adding them to this secret :)

Reply:
Haha, fair.

The controller reads these secret values and injects them into the CP template when doing its reconciliation. This [cpOpt](https://github.com/aws/eks-anywhere/pull/3424/files#diff-42413ee5a1c110d438032dcbc28aa33f9af26f3235f367868883b4b5b3447579R200) string map is eventually injected into `template-cp.yaml` by the controller.

Reply:
The request against the secret happens here: https://github.com/aws/eks-anywhere/pull/3424/files#diff-42413ee5a1c110d438032dcbc28aa33f9af26f3235f367868883b4b5b3447579R190

Reply:
I see. The problem is existing clusters won't have these values in their secret, so the controller will fail reconciling. I fear we might need to read to other secrets for backwards compatibility.

Reply:
Right, of course. I'll default to the `username` and `password` values if the others aren't present.